build/tools/Automation/DotNet/DotNet.cs

```diff
@@ -0,0 +1,236 @@
+                    sha = GetSha(release.AspnetCoreRuntime.Files);
+                    platformConstant = new PlatformConstant
+                    {
+                        Version = aspnetCoreRuntimeVersion,
+                        Sha = sha,
+                        PlatformName = Constants.DotNetName,
+                        VersionType = Constants.DotNetAspCoreName,
+                    };
+                    platformConstants.Add(platformConstant);
+
+                    // TODO: add new Major.Minor version string to runtime-version list of runtimes
+                    // for the constants.yaml list
+                    // Example: https://github.com/microsoft/Oryx/pull/1560/files#diff-47c28d7a6c8135707f46b624b5913e35beea6dfbe7a8be2db7efefde606eba59R47
+                }
+            }
+
+            return platformConstants;
+        }
+
+        /// <inheritdoc/>
+        public override void UpdateConstants(List<PlatformConstant> platformConstants, List<Constant> yamlConstants)
+        {
+            Dictionary<string, Constant> dotnetYamlConstants = GetYamlDotNetConstants(yamlConstants);
+
+            // update dotnetcore sdks and runtimes
+            foreach (var platformConstant in platformConstants)
+            {
+                string version = platformConstant.Version;
+                string sha = platformConstant.Sha;
+                string versionType = platformConstant.VersionType;
+                string dotNetConstantKey = GenerateDotNetConstantKey(platformConstant);
+                Console.WriteLine($"version: {version} versionType: {versionType} sha: {sha} dotNetConstantKey: {dotNetConstantKey}");
+                if (versionType.Equals(Constants.SdkName))
+                {
+                    Constant dotNetYamlConstant = dotnetYamlConstants[Constants.DotNetSdkKey];
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // add sdk to versionsToBuild.txt
+                    UpdateVersionsToBuildTxt(platformConstant);
+                }
+                else
+                {
+                    Constant dotNetYamlConstant = dotnetYamlConstants[Constants.DotNetRuntimeKey];
+                    dotNetYamlConstant.Constants[dotNetConstantKey] = version;
+
+                    // store SHAs for net-core and aspnet-core
+                    dotNetYamlConstant.Constants[$"{dotNetConstantKey}-sha"] = sha;
+                }
+            }
+
+            var serializer = new SerializerBuilder()
+                .WithNamingConvention(UnderscoredNamingConvention.Instance)
+                .Build();
+
+            var stringResult = serializer.Serialize(yamlConstants);
+            File.WriteAllText(Constants.ConstantsYaml, stringResult);
+        }
+
+        private static void UpdateVersionsToBuildTxt(PlatformConstant platformConstant)
+        {
+            List<string> versionsToBuildTxtFiles = new List<string>() {
```

Comment:
Similar comment to one I have elsewhere about having file paths and using that with `File.ReadAll*` methods